contents/javascript/Javascript-framework-VueJs.md

```diff
@@ -228,6 +234,24 @@ In this case, only the `root` component can be accessed in Vue while the rest ar
 
 <br>
 
+7. **Lifecycle Hooks**<br>
+    Every Vue instance goes through a series of initialization steps when it is created, i.e. setting up data observation, compiling the template, mounting the instance to the DOM, and updating the DOM when data modifies. Along these steps, Vue runs functions in the background called lifecycle hooks, allowing users to add code at each stage that could improve its rendering speed.<br/><br/>
+    The following diagram shows all lifecycle hooks and their specific execution stages:<br/>
+    ![Vue's Lifecycle Diagram](javascript-framework-vue-images/vue-lifecycle-diagram.png "Vue's Lifecycle Diagram")
```

Comment:
The image is a little blurry here. Could resave it at a higher resolution perhaps